File: src/Tankbuch.Contracts/Dtos.cs
namespace Tankbuch.Contracts;

// ---------- Auth ----------
public record RequestCodeRequest(string Email);
public record RequestCodeResponse(string Message, string? DemoCode);
public record VerifyCodeRequest(string Email, string Code);
public record VerifyCodeResponse(string Token, string Email, Guid TenantId, string TenantName);
public record MeResponse(string Email, Guid TenantId, string TenantName);

// ---------- Fahrzeuge ----------
public record FahrzeugDto(
    Guid Id,
    string Name,
    string Kennzeichen,
    string Kraftstoffart,
    string Farbe,
    long AnfangsKilometer,
    long AktuellerKilometerstand,
    int AnzahlTankvorgaenge,
    decimal? DurchschnittsVerbrauch,
    decimal Gesamtkosten);

public record CreateFahrzeugRequest(
    string Name,
    string? Kennzeichen,
    string Kraftstoffart,
    string Farbe,
    long AnfangsKilometer);

public record UpdateFahrzeugRequest(
    string Name,
    string? Kennzeichen,
    string Kraftstoffart,
    string Farbe,
    long AnfangsKilometer);

// ---------- Tankvorgänge ----------
public record TankvorgangDto(
    Guid Id,
    Guid FahrzeugId,
    DateOnly Datum,
    decimal Liter,
    decimal PreisProLiter,
    decimal Gesamtpreis,
    long Kilometerstand,
    string? Tankstelle,
    string? Notiz,
    bool Volltankung,
    decimal? Verbrauch);

public record CreateTankvorgangRequest(
    Guid FahrzeugId,
    DateOnly Datum,
    decimal Liter,
    decimal? PreisProLiter,
    decimal Gesamtpreis,
    long Kilometerstand,
    string? Tankstelle,
    string? Notiz,
    bool Volltankung);

public record UpdateTankvorgangRequest(
    DateOnly Datum,
    decimal Liter,
    decimal? PreisProLiter,
    decimal Gesamtpreis,
    long Kilometerstand,
    string? Tankstelle,
    string? Notiz,
    bool Volltankung);

// ---------- Statistik ----------
public record StatistikDto(
    string Auswahl,
    decimal Gesamtliter,
    decimal Gesamtkosten,
    decimal? DurchschnittsVerbrauch,
    decimal? DurchschnittsPreis,
    long GefahreneKilometer,
    decimal? KostenProKm,
    decimal? KostenProMonat,
    int AnzahlTankvorgaenge,
    TankvorgangDto? LetzterTankvorgang);

// ---------- OCR ----------
public record PumpOcrResult(decimal? Liter, decimal? Gesamtpreis, decimal? PreisProLiter, bool Simuliert);
public record TachoOcrResult(long? Kilometerstand, bool Simuliert);

// ---------- CSV ----------
public record CsvImportResult(int Importiert, int Uebersprungen, int NeueFahrzeuge, string Meldung);

File: ios/Tankbuch/Models/Dtos.swift
import Foundation

// Codable-Spiegel der Contracts-DTOs (API-JSON = camelCase, DateOnly = "yyyy-MM-dd").

struct Fahrzeug: Codable, Identifiable, Hashable {
    let id: UUID
    var name: String
    var kennzeichen: String
    var kraftstoffart: String
    var farbe: String
    var anfangsKilometer: Int64
    var aktuellerKilometerstand: Int64
    var anzahlTankvorgaenge: Int
    var durchschnittsVerbrauch: Double?
    var gesamtkosten: Double
}

struct Tankvorgang: Codable, Identifiable, Hashable {
    let id: UUID
    var fahrzeugId: UUID
    var datum: String // yyyy-MM-dd – String wie im Web, damit Sortierung/Vergleiche identisch bleiben
    var liter: Double
    var preisProLiter: Double
    var gesamtpreis: Double
    var kilometerstand: Int64
    var tankstelle: String?
    var notiz: String?
    var volltankung: Bool
    var verbrauch: Double?
}

struct RequestCodeResponse: Codable {
    let message: String
    let demoCode: String?
}

struct VerifyCodeResponse: Codable {
    let token: String
    let email: String
    let tenantId: UUID
    let tenantName: String
}

struct MeResponse: Codable {
    let email: String
    let tenantId: UUID
    let tenantName: String
}

struct PumpOcrResult: Codable {
    let liter: Double?
    let gesamtpreis: Double?
    let preisProLiter: Double?
    let simuliert: Bool
    let meldung: String?
}

struct TachoOcrResult: Codable {
    let kilometerstand: Int64?
    let simuliert: Bool
    let meldung: String?
}

struct CsvImportResult: Codable {
    let importiert: Int
    let uebersprungen: Int
    let neueFahrzeuge: Int
    let meldung: String
}

// Request-Bodies (synthetisiertes Encodable lässt nil-Optionals weg)

struct FahrzeugBody: Encodable {
    var name: String
    var kennzeichen: String
    var kraftstoffart: String
    var farbe: String
    var anfangsKilometer: Int64
}

struct TankvorgangBody: Encodable {
    var fahrzeugId: UUID? // nur beim Anlegen
    var datum: String
    var liter: Double
    var preisProLiter: Double?
    var gesamtpreis: Double
    var kilometerstand: Int64
    var tankstelle: String
    var notiz: String
    var volltankung: Bool
}
